hi-ml/testhiml/testhiml/test_run_ml.py

```diff
@@ -408,45 +446,45 @@ def test_run(run_inference_only: bool, run_extra_val_epoch: bool, ml_runner_with
 
 
 @pytest.mark.parametrize("run_extra_val_epoch", [True, False])
-def test_run_inference_only(run_extra_val_epoch: bool, ml_runner_with_run_id: MLRunner) -> None:
+def test_run_inference_only(run_extra_val_epoch: bool, ml_runner_with_src_ckpt: TrainingRunner) -> None:
```

Comment:
Would it make more sense for this to live in `test_eval_runner.py`? Is there actually  a use case for creating a TrainingRunner when run_inference_only = True?

Reply:
I think that would be a good next step to refactor. We should introduce more modes and delete the "run_inference_only" flag